p2p/src/peers/mod.rs

```diff
@@ -179,6 +189,17 @@ impl Peers {
         Ok(bucket.values().nth(index).map(|v| v.address.to_owned()))
     }
 
+    /// Get a random socket address from the new peers list
+    pub fn get_new_random(&self) -> Option<(u16, SocketAddr)> {
+        // Random index with range [0, len) of the peers vector
+        let index = thread_rng().gen_range(0, self.new_bucket.len());
+
+        self.new_bucket
+            .iter()
+            .nth(index)
+            .map(|(k, v)| (*k, v.address.to_owned()))
+    }
+
```

Comment:
Please note the existence of this:
https://docs.rs/rand/0.7.0/rand/seq/trait.IteratorRandom.html

Reply:
We do the operation of "chose a random element from a hashmap" many times, perhaps we can create an issue for this.

Reply:
oks, we can include in a new issue

Reply:
Shall we take the chance to apply the change here as an example that others can follow?

Reply:
Ping.